src/main/java/seedu/address/model/person/DaysAttended.java

```diff
@@ -70,25 +70,31 @@ public int hashCode() {
 
     /**
      * Increments the days attended by 1.
+     *
+     * @return A new {@code DaysAttended} instance representing the updated days attended.
      */
-    public void increment() {
-        daysAttended.set(getDaysAttended() + 1);
+    public DaysAttended incremented() {
+        return new DaysAttended(getValue() + 1);
     }
 
     /**
-     * Decrements the days attended by 1.
+     * Decrements the days attended by 1, ensuring it does not fall below 0.
+     *
+     * @return A new {@code DaysAttended} instance representing the updated days attended,
+     *         or 0 if the current days attended is 0.
      */
-    public void decrement() {
-        if (getDaysAttended() > 0) {
-            daysAttended.set(getDaysAttended() - 1);
-        }
+    public DaysAttended decremented() {
+        int newDays = Math.max(getValue() - 1, 0);
```

Comment:
Good check for ensuring minimum value of DaysAttended is always greater than or equal to zero.